x/evm/types/params.go

```diff
@@ -153,10 +146,23 @@ func validateEIPs(i interface{}) error {
 		return fmt.Errorf("invalid EIP slice type: %T", i)
 	}
 
+	uniqueEIPs := make(map[int64]struct{})
+	duplicateEIPs := make([]int64, 0, len(eips))
+
 	for _, eip := range eips {
 		if !vm.ValidEip(int(eip)) {
 			return fmt.Errorf("EIP %d is not activateable, valid EIPS are: %s", eip, vm.ActivateableEips())
 		}
+
+		if _, ok := uniqueEIPs[eip]; ok {
+			duplicateEIPs = append(duplicateEIPs, eip)
+		} else {
+			uniqueEIPs[eip] = struct{}{}
+		}
+	}
+
+	if len(duplicateEIPs) != 0 {
+		return fmt.Errorf("found duplicate EIPs: %v", duplicateEIPs)
 	}
```

Comment:
not needed, return early as soon as a dup entry has been found

Reply:
IMHO this is way better UX